iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp

```diff
@@ -54,19 +54,19 @@ static void memsetSigbusHandler(int) noexcept
     _exit(EXIT_FAILURE);
 }
 
+// NOLINTJUSTIFICATION the function size is related to the error handling and the cognitive complexity
+// results from the expanded log macro
+// NOLINTNEXTLINE(readability-function-size,readability-function-cognitive-complexity)
 cxx::expected<SharedMemoryObject, SharedMemoryObjectError> SharedMemoryObjectBuilder::create() noexcept
 {
     auto printErrorDetails = [this] {
         LogError() << "Unable to create a shared memory object with the following properties [ name = " << m_name
                    << ", sizeInBytes = " << m_memorySizeInBytes << ", access mode = " << asStringLiteral(m_accessMode)
-                   << ", open mode = " << asStringLiteral(m_openMode)
-                   << ", baseAddressHint = "
-                   // NOLINTJUSTIFICATION baseAddressHint printed on failure to make debugging easier as uint64 hex
-                   // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
-                   << ((m_baseAddressHint) ? log::HexFormat(reinterpret_cast<uint64_t>(*m_baseAddressHint))
-                                           : log::HexFormat(static_cast<uint64_t>(0U)))
+                   << ", open mode = " << asStringLiteral(m_openMode) << ", baseAddressHint = "
+                   << ((m_baseAddressHint) ? iox::log::hex(m_baseAddressHint.value())
+                                           : iox::log::hex(static_cast<uint64_t>(0U)))
                    << ((m_baseAddressHint) ? "" : " (no hint set)")
-                   << ", permissions = " << log::BinFormat(static_cast<mode_t>(m_permissions)) << " ]";
+                   << ", permissions = " << iox::log::hex(static_cast<mode_t>(m_permissions)) << " ]";
```

Comment:
This is actually wrong, we would like to have `iox::log::oct` here to see permissions like `750` and hex format is completely confusing.
Could we add this to some issue and as a compromise we just print out the good old decimal number?

Reply:
fixed in #1621